src/types/simple/serialize.rs

```diff
@@ -65,19 +65,27 @@ pub(crate) enum SerSimpleType {
     },
 }
 
-impl<T: PrimType + Into<SimpleType>> From<Container<T>> for SerSimpleType {
+impl PrimType for SerSimpleType {
+    const LINEAR: bool = true;
```

Comment:
This is one of the downsides of parameterizing `TypeRow<T: PrimType>` and then trying to instantiate with `SerSimpleType` :-(

Another possibility is to remove `const LINEAR` and require instance `.is_linear(&self) -> bool` which can be implemented by recursive traversal for SimpleType (being cautious! Or inspecting only one level if one is less cautious), by constant-false for ClassicType, and by panicking for SerSimpleType...